ebi_eva_common_pyutils/spring_properties.py

```diff
@@ -26,90 +28,153 @@ def __init__(self, maven_profile, private_settings_file):
+            {
+                'spring.batch.job.names': 'CREATE_SUBSNP_ACCESSION_JOB',
+                'parameters.assemblyReportUrl': self._format_str('file:{0}', assembly_report),
+                'parameters.chunkSize': 100,
+                'parameters.contigNaming': 'NO_REPLACEMENT',
+                'parameters.fasta': fasta,
+                'parameters.forceRestart': 'false',
+                'parameters.projectAccession': project_accession,
+                'parameters.taxonomyAccession': taxonomy_accession,
+                'parameters.vcfAggregation': aggregation,
+                'parameters.vcf': vcf_file,
+                'parameters.outputVcf': output_vcf
+            },
+        )
 
     def get_remapping_extraction_properties(self, *, taxonomy='', source_assembly='', fasta='', assembly_report='',
                                             projects='', output_folder='.'):
         """Properties for remapping extraction pipeline."""
-        return self._common_properties(source_assembly) + f'''
-spring.batch.job.names=EXPORT_SUBMITTED_VARIANTS_JOB
-parameters.taxonomy={taxonomy}
-parameters.fasta={fasta}
-parameters.assemblyReportUrl={'file:' if assembly_report else ''}{assembly_report}
-parameters.projects={projects}
-parameters.outputFolder={output_folder}
-'''
+        return self._format(
+            self._common_properties(source_assembly),
+            {
+                'spring.batch.job.names': 'EXPORT_SUBMITTED_VARIANTS_JOB',
+                'parameters.taxonomy': taxonomy,
+                'parameters.fasta': fasta,
+                'parameters.assemblyReportUrl': self._format_str('file:{0}', assembly_report),
+                'parameters.projects': projects,
+                'parameters.outputFolder': output_folder
+            })
 
     def get_remapping_ingestion_properties(self, *, source_assembly='', target_assembly='', vcf='', load_to='',
                                            remapping_version=1.0):
         """Properties for remapping ingestion pipeline."""
-        return self._common_properties(target_assembly) + f'''
-spring.batch.job.names=INGEST_REMAPPED_VARIANTS_FROM_VCF_JOB
-parameters.vcf={vcf}
-parameters.remappedFrom={source_assembly}
-parameters.loadTo={load_to}
-parameters.remappingVersion={remapping_version}
-'''
+        return self._format(
+            self._common_properties(target_assembly),
+            {
+                'spring.batch.job.names': 'INGEST_REMAPPED_VARIANTS_FROM_VCF_JOB',
+                'parameters.vcf': vcf,
+                'parameters.remappedFrom': source_assembly,
+                'parameters.loadTo': load_to,
+                'parameters.remappingVersion': remapping_version,
+            }
+        )
 
     def get_clustering_properties(self, *, instance,
                                   job_name='', target_assembly='', rs_report_path='', projects='',
-                                  project_accession='', vcf='', source_assembly=''):
```

Comment:
The clustering pipelines actually do require source assembly / `remappedFrom`, though only processing of remapped variants and backpropagation use it.